src/WWTelematics.TenantService/Application/TenantManager.cs

```
@@ -184,6 +184,71 @@ public ApplicationLoadBalancerResponse CreateTenant(string tenantRequest)
+        /// Update a single tenant by the PublicId and returns ALB response.
+        /// </summary>
+        /// <param name="publicId">Public id of the tenant to retrieve.</param>
+        /// <param name="tenantRequest">The tenant to update.</param>
+        /// <returns>An ALB Response.</returns>
+        public ApplicationLoadBalancerResponse UpdateTenantByPublicId(Guid publicId, string tenantRequest)
+        {
+            this.logger.LogTrace("Updating the tenant.");
+            TenantWrapper tenantWrapper;
+
+            if (string.IsNullOrEmpty(tenantRequest))
+            {
+                return ALBResponseHelper.GetDefaultALBResponse(400);
+            }
+
+            try
+            {
+                tenantWrapper = JsonConvert.DeserializeObject<TenantWrapper>(tenantRequest);
+                if (tenantWrapper.Tenant == null)
+                {
+                    var error = new JsonSerializationException("Unable to parse the request body. Please submit in correct format.");
+                    return ALBResponseHelper.GetDefaultALBResponse(400, error.ToString());
+                }
+            }
+            catch (JsonException)
+            {
+                var error = new JsonSerializationException("Unable to parse the request body. Please submit request in Json format.");
+                return ALBResponseHelper.GetDefaultALBResponse(400, error.ToString());
+            }
+
+            Tenant tenantUpdate = this.tenantContext.Tenants.Where(p => p.PublicId.Equals(publicId)).FirstOrDefault();
+
+            if (tenantUpdate == null)
+            {
+                return ALBResponseHelper.GetDefaultALBResponse(404, "Tenant does not exist.");
+            }
+
+            tenantUpdate.Label = !string.IsNullOrEmpty(tenantWrapper.Tenant.Label) ? tenantWrapper.Tenant.Label : tenantUpdate.Label;
+            tenantUpdate.LicenseCount = tenantWrapper.Tenant.LicenseCount > 0 ? tenantWrapper.Tenant.LicenseCount : tenantUpdate.LicenseCount;
```

The input should be validated before assignment.

Attempting to set LicenseCount to -1 should return an error.